pkg/k8sclient/cluster.go

```diff
@@ -126,35 +126,29 @@ func (g *Cluster) GetAWSClusterRoleIdentity(ctx context.Context, namespacedName
 }
 
 func (g *Cluster) HasStatusCondition(ctx context.Context, cluster *capi.Cluster) bool {
-	for _, c := range cluster.Status.Conditions {
-		if c.Type == networkTopologyCondition {
-			return true
-		}
-	}
-	return false
+	_, found := lookupConditionOrCreateUnknown(ctx, cluster, networkTopologyCondition)
+	return found
 }
 
 func (g *Cluster) UpdateStatusCondition(ctx context.Context, cluster *capi.Cluster, status corev1.ConditionStatus) error {
```

Comment:
There are these helper funcs in CAPI to update conditions in the CR, `MarkTrue(...)`, `MarkUnknown(...)`, `MarkFalse(...)`, with appropriate/required args (and internal implementation) that obeys upstream proposal. These funcs construct the `Condition` object based on passed args.

Check it here https://github.com/kubernetes-sigs/cluster-api/blob/main/util/conditions/setter.go#L109-L122

So you should be able to just use those funcs to update the CR in-memory, and then just client.Patch/Update it in our code.

The benefit of using upstream `Mark*` funcs is that they should/will get updated accordingly if/when upstream conditions proposal and specs change (the current proposal is not final, so I believe it will change, possibly to match Kubernetes conditions). So less code on our side to maintain and update :)